logic/src/commonMain/kotlin/com/wire/kalium/logic/data/message/Message.kt

```diff
@@ -344,7 +344,9 @@ data class MessagePreview(
     val content: MessagePreviewContent,
     val date: String,
     val visibility: Message.Visibility,
-    val isSelfMessage: Boolean
+    val isSelfMessage: Boolean,
+    val senderUserId: UserId,
+    val selfUserId: UserId?
```

Comment:
If we pass both `senderUserId` and `selfUserId` then `isSelfMessage` is just `senderUserId == selfUserId`, right? So it's redundant 😄  
I'm not sure if we should add `selfUserId` here to `MessagePreview` because it's not actually a part of message data, especially if we use that only for `MemberChange` system messages. 
For regular messages, list of user ids is passed so it's exactly like now in `MessagePreviewEntityContent.MemberChange` and the details of each user for UI are taken using `ObserveUserListByIdUseCase`, so you can get all user names that were added/removed and check whether any of them is the `selfUserId`, because from this use case we get list of `User` which can be either of type `OtherUser` or `SelfUser`. If we don't need the names and all of that, then maybe it would still be better to pass `selfUserId` later in AR since it's only to display the text properly, what do you think? 😄

Reply:
I changed to have this information already in entity class